### PR TITLE
Co mrc file formats

### PR DESCRIPTION
**Pull request by cossorzano**

### CODE REVIEW

File: core/xmipp_image_base.cpp [hunk 1]
```diff
@@ -883,9 +912,12 @@ int ImageBase::_readBatch(const FileName &name, ImageFHandler* hFile, size_t sta
     if (ext_name.contains("spi") || ext_name.contains("xmp")  ||
         ext_name.contains("stk") || ext_name.contains("vol")) {
         err = readSPIDER(start_img, batch_size);
-    } else if (ext_name.contains("mrcs")||ext_name.contains("st")) { //mrc stack MUST go BEFORE plain MRC
+    } else if (ext_name.contains("mrcs") || ext_name.contains("st") ||
```
Comment (DStrelak) [inline core/xmipp_image_base.cpp:915]: duplicated code. Extract methods such as
`isMRCType(ext)` etc.
Don't forget to call that method in the write methods as well

File: core/xmipp_image_base.cpp [hunk 2]
```diff
@@ -1014,10 +1047,11 @@ void ImageBase::_write(const FileName &name, ImageFHandler* hFile, size_t select
         err = writeSPIDER(select_img,isStack,mode);
     else if (ext_name.contains("stk"))
         err = writeSPIDER(select_img,true,mode);
-    //    else if (ext_name.contains("mrcs"))
-    //        writeMRC(select_img,true,mode,imParam,castMode);
-    else if (ext_name.contains("mrc")||ext_name.contains("map")
-             ||ext_name.contains("mrcs")||ext_name.contains("st"))
+    else if (ext_name.contains("mrc") || ext_name.contains("map") ||
```
Comment (DStrelak) [inline core/xmipp_image_base.cpp:1050]: duplicated code, see my comment above

**Review by DStrelak — CHANGES_REQUESTED** [review]

File: core/xmipp_image_base.cpp [hunk 3]
```diff
@@ -742,6 +744,30 @@ void ImageBase::closeFile(ImageFHandler* hFile) const
     delete hFile;
 }
 
+bool isDynamicMRCS(const String &ext)
+{
+    FileName mrcStackExtensions;
+    if (getenv("XMIPP_MRC_STACK_EXTENSIONS"))
+    	mrcStackExtensions=getenv("XMIPP_MRC_STACK_EXTENSIONS");
+    String plainExt = ext;
```
[the remark this answers is not in the document]
Comment (cossorzano) [reply core/xmipp_image_base.cpp:752]: At this point, I don't have a clear vision of all possibilities. So, I prefer not to give examples.

File: core/xmipp_image_base.cpp [hunk 4]
```diff
@@ -742,6 +744,30 @@ void ImageBase::closeFile(ImageFHandler* hFile) const
     delete hFile;
 }
 
+bool isDynamicMRCS(const String &ext)
+{
+    FileName mrcStackExtensions;
+    if (getenv("XMIPP_MRC_STACK_EXTENSIONS"))
+    	mrcStackExtensions=getenv("XMIPP_MRC_STACK_EXTENSIONS");
```
[the remark this answers is not in the document]
Comment (cossorzano) [reply core/xmipp_image_base.cpp:751]: OK

File: core/xmipp_image_base.cpp [hunk 5]
```diff
@@ -742,6 +744,30 @@ void ImageBase::closeFile(ImageFHandler* hFile) const
...
+    String plainExt = ext;
+    size_t found = ext.find_first_of("%");
+    if (found!=String::npos)
+        plainExt = ext.substr(0, found) ;
+    return mrcStackExtensions.contains(plainExt);
+}
+
+bool isDynamicMRC(const String &ext)
+{
+    FileName mrcExtensions;
```
[the remark this answers is not in the document]
Comment (cossorzano) [reply core/xmipp_image_base.cpp]: Done

File: core/xmipp_image_base.cpp (hunk 1 above)
Comment (cossorzano) [reply core/xmipp_image_base.cpp]: Done

File: core/xmipp_image_base.cpp (hunk 2 above)
Comment (cossorzano) [reply core/xmipp_image_base.cpp]: Done

**Review by DStrelak — APPROVED** [review]

**Pull request merged**